templates/openstackclient/bin/tripleo-deploy.sh

```diff
@@ -2,21 +2,28 @@
 
 set -eux
 
-# disable running dhcp on all interfaces, setting disable_configure_safe_defaults in the interface template does not work
-sudo sed -i '/^set -eux/a disable_configure_safe_defaults=true' /usr/share/openstack-tripleo-heat-templates/network/scripts/run-os-net-config.sh
+render() {
+  # disable running dhcp on all interfaces, setting disable_configure_safe_defaults in the interface template does not work
+  sudo sed -i '/^set -eux/a disable_configure_safe_defaults=true' /usr/share/openstack-tripleo-heat-templates/network/scripts/run-os-net-config.sh
 
-mkdir -p ~/tripleo-deploy
-rm -rf ~/tripleo-deploy/overcloud-ansible*
-unset OS_CLOUD
+  rm -rf /home/cloud-admin/tripleo-deploy/overcloud-ansible*
+  if [ ! -L /var/log/validations ]; then
+    sudo ln -s /home/cloud-admin/tripleo-deploy/validations /var/log/validations
+  fi
 
-sudo openstack tripleo deploy \
+  unset OS_CLOUD
+
+  sudo openstack tripleo deploy \
     --templates /usr/share/openstack-tripleo-heat-templates \
-    -r /usr/share/openstack-tripleo-heat-templates/roles_data.yaml \
+    -r ROLESFILE \
     -n /usr/share/openstack-tripleo-heat-templates/network_data.yaml \
     -e /usr/share/openstack-tripleo-heat-templates/overcloud-resource-registry-puppet.yaml \
     -e /usr/share/openstack-tripleo-heat-templates/environments/network-isolation.yaml \
     -e /usr/share/openstack-tripleo-heat-templates/environments/deployed-server-environment.yaml \
     -e /usr/share/openstack-tripleo-heat-templates/environments/docker-ha.yaml \
+{{- if .EnableCeph }}
```

Comment:
Would it be reasonable to have the user just supply this via the custom-config ConfigMap instead? My concern is that logic like this may not be scalable as we add features.

Reply:
yes we can do that, then we won't need the isHCI at all and required the user to pass all information from that ceph env file into the custom-config ConfigMap, or it might be a good idea to provide an interface a user can use to specify which ../environments/.. files provided by triplo should be added to the deploy script?

Reply:
Yes. I think the OSPlaybookGenerator might be a good place for that.